File: Backend/src/main/java/com/example/project/entity/Product.java
package com.example.project.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "Product")
public class Product {
    @Id
    private int product_id;
    private String title;

    @OneToOne
    @JoinColumn(name = "image_id", referencedColumnName = "image_id")
    private Image product_image;
    private String description;

    @Column(name = "price", precision = 18, scale = 2)
    private BigDecimal price;

    @Column(name = "discount_price", precision = 18, scale = 2)
    private BigDecimal discountPrice;

    private Integer quantity_of_available;
    private String unit;

    @Column(name = "delivery_days")
    private Integer deliveryDays;

    @OneToOne(mappedBy = "product_category")
    private CategoryProduct categories;

    @OneToMany(mappedBy = "product")
    private List<OrderedProduct> product_orderes;

}

File: Android/app/src/main/java/com/kire/market_place_android/presentation/model/product/Product.kt
package com.kire.market_place_android.presentation.model.product

import java.math.BigDecimal

/**
 * @author Aleksey Timko (de4ltt)*/
data class Product(
    val id: Int = -1,
    val title: String = "",
    val image: String = "",
    val description: String = "",
    val price: BigDecimal = 0.00.toBigDecimal(),
    val discountPrice: BigDecimal = 0.00.toBigDecimal(),
    val quantityAvailable: Int = 0,
    val unit: String = "",
    val deliveryDays: Int = 0,
    val category: String = "",
    val chosenQuantity: Int = 0
)